kvbc/test/sparse_merkle_storage/db_adapter_unit_test.cpp

```diff
@@ -1189,6 +1200,135 @@ TEST_P(db_adapter_custom_blockchain, delete_genesis_and_verify_stale_index) {
   ASSERT_FALSE(hasProvableStaleIndexKeysSince(adapter.getDb(), 2));
 }
 
+TEST_P(db_adapter_custom_blockchain, delete_genesis_and_verify_stale_key_deletion_for_non_provable_keys) {
+  const Sliver skey1{"key1"};
+  const Sliver skey2{"key2"};
+  concord::kvbc::v2MerkleTree::DBAdapter::NonProvableKeySet non_provable_key_set = {skey1, skey2};
+  auto adapter = DBAdapter{GetParam()->newEmptyDb(), true, non_provable_key_set};
+
+  const auto key1 = Sliver{"k1"};
+  const auto key2 = Sliver{"k2"};
+
+  const auto ver1StaleLeafKey1 = DBKeyManipulator::genDataDbKey(key1, 1);
+  const auto ver1StaleLeafKey2 = DBKeyManipulator::genDataDbKey(key2, 1);
+
+  const auto ver1StaleNonProvableKey1 = DBKeyManipulator::genNonProvableDbKey(1, skey1);
+  const auto ver1StaleNonProvableKey2 = DBKeyManipulator::genNonProvableDbKey(1, skey2);
+
+  // Add blocks with overlapping keys. Include an empty block.
+  ASSERT_EQ(adapter.addBlock(SetOfKeyValuePairs{std::make_pair(key1, Sliver{"k1v1"}),
+                                                std::make_pair(key2, Sliver{"k2v1"}),
+                                                std::make_pair(skey1, Sliver{"sk1v1"}),
+                                                std::make_pair(skey2, Sliver{"sk2v1"})}),
+            1);
+  ASSERT_EQ(adapter.addBlock(SetOfKeyValuePairs{}), 2);
+  ASSERT_EQ(adapter.addBlock(SetOfKeyValuePairs{std::make_pair(key1, Sliver{"k1v3"}),
+                                                std::make_pair(key2, Sliver{"k2v3"}),
+                                                std::make_pair(skey1, Sliver{"sk1v2"}),
+                                                std::make_pair(skey2, Sliver{"sk2v2"})}),
+            3);
+  ASSERT_EQ(adapter.addBlock(SetOfKeyValuePairs{std::make_pair(key1, Sliver{"k1v4"}),
+                                                std::make_pair(key2, Sliver{"k2v4"}),
+                                                std::make_pair(skey1, Sliver{"sk1v3"}),
+                                                std::make_pair(skey2, Sliver{"sk2v3"})}),
+            4);
+
+  // Verify that the leaf key for key1 at version 1 is available.
+  ASSERT_TRUE(adapter.getDb()->has(ver1StaleLeafKey1).isOK());
+
+  // Verify that the leaf key for key2 at version 1 is available.
+  ASSERT_TRUE(adapter.getDb()->has(ver1StaleLeafKey2).isOK());
+  //
+  // Verify that the non-provable key for skey1 at block 1 is available.
```

Comment:
I think that comment is incorrect and it is equivalent to a check for `ver1StaleLeafKey1`. Also applies to the line below.